src/main/java/Duke.java

```diff
@@ -33,31 +33,82 @@ public void handleCommand() {
             System.out.println("-------------------------------------");
             handleCommand();
             break;
+        case "deadline":
+            System.out.println("-------------------------------------");
+            System.out.println("Got it. I've added this task:");
+            addDeadlineOrEventTask(inputWords, "deadline");
+            System.out.println("-------------------------------------");
+            handleCommand();
+            break;
+        case "event":
+            System.out.println("-------------------------------------");
+            System.out.println("Got it. I've added this task:");
+            addDeadlineOrEventTask(inputWords, "event");
+            System.out.println("-------------------------------------");
+            handleCommand();
+            break;
         default:
             System.out.println("-------------------------------------");
-            addTask(userInput);
-            System.out.println("added: " + userInput);
+            System.out.println("Got it. I've added this task:");
+            addTodoTask(inputWords);
             System.out.println("-------------------------------------");
             handleCommand();
             break;
         }
     }
 
-    public void addTask(String task) {
-        userList[listSize] = new Task(task);
+    public void addDeadlineOrEventTask(String[] inputWords, String type) {
+        // Find separator index
+        int separatorIndex = -1;
+        for (int i = 1; i < inputWords.length; i++) {
+            if (inputWords[i].equals("/by") || inputWords[i].equals("/at")) {
+                separatorIndex = i;
```

Comment:
Probably you can substitute this for loop with a startswith() to simplify the code.